### PR TITLE
Add container with packer and ansible

### PR DESCRIPTION
**Pull request by awangc**
For running in concourse, a docker image with both packer and ansible

### CODE REVIEW

File: packer-ansible/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM hashicorp/packer
+LABEL maintainer="williamofockham <occam_engineering@comcast.com>"
+
+ARG ANSIBLE_VERSION
+
+RUN set -x && \
```
Comment (drunkirishcoder) [inline packer-ansible/Dockerfile]: can we move all the `&&` to the beginning of next line?

File: packer-ansible/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM hashicorp/packer
```
Comment (drunkirishcoder) [inline packer-ansible/Dockerfile]: Typically I like to pin to a specific image tag. Not a big deal for tooling I guess. Just a little more more predictable than `latest`.

File: packer-ansible/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      gcc \
+      musl-dev \
+      libffi-dev \
+      openssl-dev \
+      make \
+      python-dev \
+    && \
+    \
+    echo "===> Installing sudo to emulate normal OS behavior..."  && \
+    apk update && apk upgrade &&\
```
Comment (drunkirishcoder) [inline packer-ansible/Dockerfile]: why updating `apk` again here? also don't see `sudo`. can you explain what's happening here?

File: packer-ansible/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    \
+    echo "==> Adding Python runtime..."  && \
+    apk add --no-cache py-pip bash && \
+    pip install --upgrade pip && \
+    echo "====> Installing ansible..." && \
+    pip install docker && \
```
Comment (drunkirishcoder) [inline packer-ansible/Dockerfile]: what's docker used for?

File: packer-ansible/Dockerfile (hunk 3 above)
Comment (awangc) [reply packer-ansible/Dockerfile]: I was trying with and without sudo and forgot to update the comments. And yes, no need to update.

File: packer-ansible/Dockerfile (hunk 4 above)
Comment (awangc) [reply packer-ansible/Dockerfile]: for ansible docker_* modules. see requirements in https://docs.ansible.com/ansible/latest/modules/docker_container_module.html.

we use those modules to configure images when building using packer.

File: packer-ansible/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    && apk add --no-cache py-pip bash          \
+    && pip install --upgrade pip               \
+    && echo "====> Installing ansible..."      \
+    && pip install docker                      \
+    && pip install ansible==${ANSIBLE_VERSION} \
+    && apk del build-dependencies              \
+    && rm -rf /var/cache/apk/*                 \
+    \
+    \
+    echo "===> Adding hosts for convenience..."  && \
```
Comment (drunkirishcoder) [inline packer-ansible/Dockerfile]: missing `&&` before `echo`, maybe remove one of the preceding `\` and two more `&&` to move to next line.

Comment (awangc) [reply packer-ansible/Dockerfile]: thanks for catching that! oh my, the dangers of `rm -rf`..

**Pull request merged**